qemu/tests/dump_guest_memory.py

```diff
@@ -0,0 +1,229 @@
+                          "Actual result: '%s'" % (expect_o, qmp_o))
+        elif result_check == "contain":
+            for o in expect_o:
+                if isinstance(o, dict):
+                    for key, val in o.items():
+                        result = check_list(qmp_o, key, val)
+                        if not result:
+                            break
+                elif isinstance(o, str):
+                    result = check_list(qmp_o, o, check_item_in_pair=False)
+
+                if result:
+                    logging.info("QMP output contain the expect value %s" % o)
+                else:
+                    test.fail("QMP output does not contain the expect value.\n"
+                              "Missed expect value: '%s'\n"
+                              "Actual result: '%s'\n" % (o, qmp_o))
+
+    def execute_qmp_cmd(qmp_cmd, expect_result):
+        """
+        Execute qmp command and check if result as expect
+
+        :param qmp_cmd: qmp command
+        :type qmp_cmd: str
+        :param expect_result: expect result of qmp command
+        :type expect_result: str
+        """
+        # qmp command
+        try:
+            # Testing command
+            logging.info("Run qmp command '%s'.", qmp_cmd)
+            qmp_o = qmp_port.send_args_cmd(qmp_cmd)
+            if not isinstance(qmp_o, list):
+                qmp_o = [qmp_o]
+            logging.debug("QMP command: '%s' \n Output: '%s'", qmp_cmd, qmp_o)
+        except Exception as err:
+            qmp_o = err.data
+            logging.info(err)
+
+        if result_check == "equal" or result_check == "contain":
+            logging.info("Verify qmp command '%s'." % qmp_cmd)
+            check_result(qmp_o, eval(expect_result))
+
+    def check_dump_file(file_name):
+        """
+        Use crash to check dump file
+
+        :param file_name: dump file name
+        :type file_name: str
+
+        """
+        try:
+            kernel_version = process.getoutput("uname -r")
+            crash_script = "/home/crash.cmd"
+            process.run("echo bt > %s" % crash_script)
+            process.run("echo quit >> %s" % crash_script)
+            crash_cmd = "crash -i %s /usr/lib/debug/lib/modules/%s/vmlinux "
+            crash_cmd %= (crash_script, kernel_version)
+            crash_cmd += file_name
+            output = process.getoutput(crash_cmd)
+            logging.debug("%s", output)
+        except process.CmdError as err:
+            output = err.result.stderr.decode()
+        if output.find("error") >= 0:
+            test.fail("vmcore corrupt")
+
+    # install crash/gdb
+    install_package("crash")
+    install_package("gdb")
+
+    qemu_binary = utils_misc.get_qemu_binary(params)
+    if not utils_misc.qemu_has_option("qmp", qemu_binary):
+        test.cancel("Host qemu does not support qmp.")
+
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+
+    session = vm.wait_for_login(timeout=int(params.get("login_timeout", 360)))
```

Comment:
Default `timewout` would be fine when you have no specific requirement for your test case, so here `vm.wait_for_login()` is fine.